platform/openharmony/export/export_plugin.cpp

```diff
@@ -300,6 +300,31 @@ Error EditorExportPlatformOpenHarmony::export_project_helper(const Ref<EditorExp
 	String project_name = p_path.get_file().get_basename();
 	String project_dir = base_dir.path_join(project_name);
 	String file_ext = p_path.get_extension();
+	bool is_windows = OS::get_singleton()->get_name() == "Windows";
+	bool is_project_dir_valid = true;
+
+	if (project_dir.is_empty() || project_dir.ends_with(".")) {
+		is_project_dir_valid = false;
+	} else {
+		Vector<String> parts = project_dir.replace("\\", "/").split("/");
+		if (is_windows) {
+			parts.remove_at(0);
+		}
```

Comment:
`OS::get_singleton()->get_name() == ` can be changed to conditional compilation.
```
#ifdef WINDOWS_ENABLED
		parts.remove_at(0);
#endif
```